up/up.go

```diff
@@ -0,0 +1,181 @@
+// Package up Copyright © 2023 ScienceLogic Inc/*
+
+package up
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/viper"
+	"github.com/zebrium/ze-cli/batch"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+)
+
+var tokenPath = "log/api/v2/token"
+var logstashPath = "log/api/v2/ingest?log_source=logstash&log_format=json_batch"
+var postPath = "log/api/v2/post"
+
+func UploadFile(url string, auth string, file string, logtype string, host string, svcgrp string,
+	dtz string, ids string, cfgs string, tags string, batchId string, disableBatch bool, logstash bool,
+	version string) (err error) {
+	route := postPath
+	client := &http.Client{}
+
+	metadata, existingBatch, err := generateMetadata(url, auth, file, logtype, host, svcgrp, dtz, ids, cfgs, tags, batchId, disableBatch, version)
+	if err != nil {
+		cleanUpBatchOnExit(batchId)
+		return err
+	}
+	body, err := json.Marshal(metadata)
+	if err != nil {
+		cleanUpBatchOnExit(batchId)
+		return err
+	}
+	//Request Stream Token
+	req, err := http.NewRequest("POST", fmt.Sprintf("%s/%s", url, tokenPath), bytes.NewBuffer(body))
+	if err != nil {
+		cleanUpBatchOnExit(batchId)
+		return err
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Token %s", auth))
+	req.Header.Add("Content-Type", "application/json")
+	resp, err := client.Do(req)
+	if err != nil {
+		cleanUpBatchOnExit(batchId)
+		return err
+	}
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		cleanUpBatchOnExit(batchId)
+		return err
+	}
+	tokenResp := &TokenResp{}
+	err = json.Unmarshal(respBody, &tokenResp)
+	if err != nil {
+		cleanUpBatchOnExit(batchId)
+		return err
+	}
+	uploadAuth := tokenResp.Token
+	if logstash {
+		route = logstashPath
+		uploadAuth = auth
+	}
+	req, err = sendRequestBuilder(url, route, file)
+
+	if err != nil {
+		cleanUpBatchOnExit(batchId)
+		return err
+	}
+
+	req.Header.Add("Authorization", fmt.Sprintf("Token %s", uploadAuth))
+	resp, err = client.Do(req)
+	if err != nil {
+		cleanUpBatchOnExit(batchId)
+		return err
+	}
+
+	//Close batch
+	if !disableBatch && !existingBatch {
```

Comment:
Here you can use a defer func at the start. 

defer func() { if err != nil {
       cleanUpBatchOnExit(batchId) } ()

Then you can do 👍 
if err != nil {
   return err
}